### PR TITLE
Use WebAssembly version libraries for unit testing

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: tests/utils/pal-wasm-testing.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    throw new Error(`Don't know how to fetch buffer at url ${binaryUrl}`);
+}
+
+export async function ensureWasmModuleReady() {
+    return Promise.resolve();
+}
+
+checkPalIntegrity<typeof import('@pal/wasm')>(withImpl<typeof import('./pal-wasm-testing')>());
```
Comment (PPpro) [inline tests/utils/pal-wasm-testing.ts:53]: why add this file ?

**Comment by PPpro** [comment]
and please remember to update the related docs https://github.com/cocos/3d-tasks/blob/master/design-docs/scripting/engine-wasm-module.md#%E8%A7%A3%E5%86%B3-ci-%E9%97%AE%E9%A2%98

File: tests/utils/pal-wasm-testing.ts (hunk 1 above)
Comment (shrinktofit) [reply tests/utils/pal-wasm-testing.ts:53]: It makes it possible to use wasm in unit test.

Comment (PPpro) [reply tests/utils/pal-wasm-testing.ts:53]: why don't actually mock pal/wasm module to the web implementation

**Review by PPpro — APPROVED** [review]

**Review by lealzhan — APPROVED** [review]

**Pull request merged**